snap/hooks/configure

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh -e
+
+set -eux
+
+# Note: the future we might want to handle scenarios where we reseting the URI restarts the daemon
+
+# Create the config file that is used by pbm
+CONFIG_FILE="$SNAP_COMMON/pbm_config.yaml"
+if [ ! -f "${CONFIG_FILE}" ]; then
+    echo "Using default config ${CONFIG_FILE}"
+    touch "$CONFIG_FILE"
+fi
+
+# set the config file for pbm
+if [ -n "$(snapctl get uri)" ]
+then
+    export PBM_MONGODB_URI="$(snapctl get uri)"
```

Comment:
Exporting and assigning variable will mask subshell non-zero exit code (you are using `set -e` above, so you want to abort on failure):

```suggestion
    PBM_MONGODB_URI=$(snapctl get uri)
    export PBM_MONGODB_URI
```

Maybe one call for snapctl is enough/faster:
```
PBM_MONGODB_URI=$(snapctl get uri)
if [ -n "${PBM_MONGODB_URI}" ]; then
   export PBM_MONGODB_URI
fi
```